index.js

```diff
@@ -277,70 +41,76 @@ function words(options) {
     return wordList[randInt(wordList.length)];
   }
 
+  // limits the size of words to the minimum and maximum possible
+  function limitWordSize(wordSize) {
+    if (wordSize < shortestWordSize) wordSize = shortestWordSize;
+    if (wordSize > longestWordSize) wordSize = longestWordSize;
+    return wordSize;
+  }
+
   // random int as seeded by options.seed if applicable, or Math.random() otherwise
   function randInt(lessThan) {
     const r = random ? random() : Math.random();
     return Math.floor(r * lessThan);
   }
 
   // No arguments = generate one word
-  if (typeof(options) === 'undefined') {
+  if (options === undefined) {
     return word();
   }
 
   // Just a number = return that many words
-  if (typeof(options) === 'number') {
+  if (typeof options === "number") {
     options = { exactly: options };
+  } else if (Object.keys(rest).length === 0) {
+    return word();
   }
 
   // options supported: exactly, min, max, join
   if (options.exactly) {
     options.min = options.exactly;
     options.max = options.exactly;
   }
-  
+
   // not a number = one word par string
-  if (typeof(options.wordsPerString) !== 'number') {
+  if (typeof options.wordsPerString !== "number") {
     options.wordsPerString = 1;
   }
 
   //not a function = returns the raw word
-  if (typeof(options.formatter) !== 'function') {
+  if (typeof options.formatter !== "function") {
     options.formatter = (word) => word;
   }
 
   //not a string = separator is a space
-  if (typeof(options.separator) !== 'string') {
-    options.separator = ' ';
+  if (typeof options.separator !== "string") {
+    options.separator = " ";
   }
 
-  var total = options.min + randInt(options.max + 1 - options.min);
-  var results = [];
-  var token = '';
-  var relativeIndex = 0;
+  const total = options.min + randInt(options.max + 1 - options.min);
+  let results = [];
+  let token = "";
+  let relativeIndex = 0;
 
-  for (var i = 0; (i < total * options.wordsPerString); i++) {
+  for (let i = 0; i < total * options.wordsPerString; i++) {
     if (relativeIndex === options.wordsPerString - 1) {
       token += options.formatter(word(), relativeIndex);
-    }
-    else {
+    } else {
       token += options.formatter(word(), relativeIndex) + options.separator;
     }
     relativeIndex++;
     if ((i + 1) % options.wordsPerString === 0) {
       results.push(token);
-      token = ''; 
+      token = "";
       relativeIndex = 0;
     }
-   
   }
-  if (typeof options.join === 'string') {
+  if (typeof options.join === "string") {
     results = results.join(options.join);
   }
 
   return results;
 }
 
-module.exports = words;
 // Export the word list as it is often useful
-words.wordList = wordList;
+export default words;
```

Comment:
The word list is no longer exported. Although I realize we are moving to ESM here already and it must already be a new major version, which allows backward compatibility breaks, we prefer that any exports be made from the root file so that developers are not told to directly import sub-files, which can lead to importing of files that are not meant to be part of a public and maintained API.

So, a second, non-default export for the wordList is called for at this level.

Reply:
Isolated tests for `minLength` and `maxLength` have been added. I also added an `AssertionError message` in the tests because I had forgotten about it. 

In the end I thought it best to move wordList[] back to index, I could do a different export, something like a barrel export or something else, but I thought it best to keep things as they are and leave everything centered in `index.js`.